Client admin

TimPigden opened the pull request:
Added clientAdmin with tests. #62
Also started documentation and rearranged some tests pursuing possible bug (will raise issue)

iravid commented:
@TimPigden Can you rebase this against master?

TimPigden commented:
sorry I'm confused. I based my changes on these files and they are legitimate updates. I ran rebase master in my branch and it just tells me I'm up-to-date. I did a cosmetic change to force a new upload, but I'm not convinced that's enough. For the conflicting files we need to take the new version from here. Or should I merge my changes into updates master and do a new PR (git is not my speciality!)

iravid commented:
Switch back to master, run `git pull`, switch back to your branch and run `git rebase master`. Then `git push -f`.
On 16 Nov 2019, 12:11 +0200, Tim Pigden <notifications@github.com>, wrote:
> sorry I'm confused. I based my changes on these files and they are legitimate updates. I ran rebase master in my branch and it just tells me I'm up-to-date. I did a cosmetic change to force a new upload, but I'm not convinced that's enough. For the conflicting files we need to take the new version from here. Or should I merge my changes into updates master and do a new PR (git is not my speciality!)
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.

TimPigden commented:
```
tim@tim-home2:~/code/zio-kafka$ git checkout master
Switched to branch 'master'
Your branch is up-to-date with 'origin/master'.
tim@tim-home2:~/code/zio-kafka$ git pull
Already up-to-date.
tim@tim-home2:~/code/zio-kafka$ git checkout client-admin
Switched to branch 'client-admin'
Your branch is up-to-date with 'origin/client-admin'.
tim@tim-home2:~/code/zio-kafka$ git rebase master
Current branch client-admin is up to date.
tim@tim-home2:~/code/zio-kafka$ git push -f
Username for 'https://github.com': tim.pigden@optrak.com
Password for 'https://tim.pigden@optrak.com@github.com':
Everything up-to-date
tim@tim-home2:~/code/zio-kafka$
```
but it appears not to do anything


On Sat, 16 Nov 2019 at 10:14, Itamar Ravid <notifications@github.com> wrote:

> Switch back to master, run `git pull`, switch back to your branch and run
> `git rebase master`. Then `git push -f`.
> On 16 Nov 2019, 12:11 +0200, Tim Pigden <notifications@github.com>, wrote:
> > sorry I'm confused. I based my changes on these files and they are
> legitimate updates. I ran rebase master in my branch and it just tells me
> I'm up-to-date. I did a cosmetic change to force a new upload, but I'm not
> convinced that's enough. For the conflicting files we need to take the new
> version from here. Or should I merge my changes into updates master and do
> a new PR (git is not my speciality!)
> > —
> > You are receiving this because you commented.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zio/zio-kafka/pull/63?email_source=notifications&email_token=AAJWHJ4SSU47DYDIGB3JHY3QT7BX5A5CNFSM4JMDXDZ2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEHOF2I#issuecomment-554623721>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJWHJYZSFCXZAIJ5EO2SC3QT7BX5ANCNFSM4JMDXDZQ>
> .
>


-- 
Tim Pigden
Optrak Distribution Software Limited
+44 (0)1992 517100
http://www.linkedin.com/in/timpigden
http://optrak.com
Optrak Distribution Software Ltd is a limited company registered in England
and Wales.
Company Registration No. 2327613 Registered Offices: Suite 6,The Maltings,
Hoe Lane, Ware, SG12 9LR England
This email and any attachments to it may be confidential and are intended
solely for the use of the individual to whom it is addressed. Any views or
opinions expressed are solely those of the author and do not necessarily
represent those of Optrak Distribution Software Ltd. If you are not the
intended recipient of this email, you must neither take any action based
upon its contents, nor copy or show it to anyone. Please contact the sender
if you believe you have received this email in error.

iravid commented:
Is `origin` pointing to your fork?

TimPigden commented:
```
 remote origin
  Fetch URL: https://github.com/TimPigden/zio-kafka.git
  Push  URL: https://github.com/TimPigden/zio-kafka.git
  HEAD branch: master
  Remote branches:
    client-admin    tracked
    feature/streams tracked
    local-compile   tracked
    master          tracked
  Local branches configured for 'git pull':
    client-admin  merges with remote client-admin
    local-compile merges with remote local-compile
    master        merges with remote master
  Local refs configured for 'git push':
    client-admin  pushes to client-admin  (up-to-date)
    local-compile pushes to local-compile (up-to-date)
    master        pushes to master        (up-to-date)
```

TimPigden commented:
```
tim@tim-home2:~/code/zio-kafka$ git remote show upstream
* remote upstream
  Fetch URL: https://github.com/zio/zio-kafka.git
  Push  URL: https://github.com/zio/zio-kafka.git
  HEAD branch: master
  Remote branches:
    feature/streams tracked
    master          tracked
  Local ref configured for 'git push':
    master pushes to master (local out of date)
tim@tim-home2:~/code/zio-kafka$ 
```

TimPigden commented:
I've looked for attempted fixes to local out of date but they just say use force - which I've already done. Everything else says "Already up-to-date" .
Shall I just give up and create a new PR?

iravid commented:
@TimPigden You can create a new PR, but you'll still have to merge the changes in master to your branch. Otherwise you'll still have conflicts.

Try doing the following:
```
git checkout master
git pull upstream master
git checkout client-admin
git rebase master
```

TapanVaishnav commented:
> Try doing the following:

shorter version just in case
```
git fetch upstream
git rebase upstream/master
git push origin client-admin --force
```

iravid commented:
Superseded by #77
